### PR TITLE
Binary versions with build metadata

### PR DESCRIPTION
**Pull request by erak**
Closes https://github.com/radicle-dev/radicle-cli/issues/239.

### CODE REVIEW

**Comment by slack-coder** [comment]
How about removing the 'vergen' package when things are finalized?  It works, but introducing a new dependency comes at a cost.

**Comment by erak** [comment]
Indeed, `vergen` comes with the cost of additional dependencies. Rebuilding this mechanism from scratch also comes with some cost though. We'd need to call a git subcommand, get the latest revision and, I think that's the tricky part, consistently create the correct Cargo config, such that the environment variable ends up being available to `rad`.

**Comment by slack-coder** [comment]
:+1: I can help out with this once we're settled on the requirements

**Comment by cloudhead** [comment]
See: https://github.com/radicle-dev/radicle-client-services/blob/master/build.rs

**Comment by slack-coder** [comment]
Here's the patch for bumping the version to [0.7.0-dev](https://github.com/slack-coder/radicle-cli/commit/84d8623413109df496c7623a97fa4f5d2e2244eb)

**Comment by erak** [comment]
@slack-coder I picked your commit, thanks! @cloudhead Together with that and the snippet you posted, the version is now being reported as e.g. `rad 0.7.0-dev+cb2c115` for development builds. If it's a release build (version does not contain `-dev`), the sha is not added.

File: cli/build.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+use std::process::Command;
+
+fn main() {
+    let hash = Command::new("git")
```
Comment (pinkforest) [inline cli/build.rs:4]: this is going to be flaky and will break things. it is better to have it in automated builds. build.rs is useful e.g. with protoc for code generation but not for managing build metadata.

**Comment by pinkforest** [comment]
fwiw - 

It's probably worthwhile to start using CHANGELOG.md using "standard" keepchangelog https://keepachangelog.com/en/1.0.0/

e.g. see example use: https://github.com/khonsulabs/bonsaidb/blob/main/CHANGELOG.md

And align on https://semver.org/spec/v2.0.0.html use. 

As to the -dev etc. suffix(es) it's best generated in -nightly, -pr etc. automated builds rather than off manifest which should reflect the "base version"

**Comment by erak** [comment]
> fwiw -
> 
> It's probably worthwhile to start using CHANGELOG.md using "standard" keepchangelog https://keepachangelog.com/en/1.0.0/
> 
> e.g. see example use: https://github.com/khonsulabs/bonsaidb/blob/main/CHANGELOG.md

Oh, yes. We should add a changelog. I've filed an issue tracking this: https://github.com/radicle-dev/radicle-cli/issues/249

> 
> And align on https://semver.org/spec/v2.0.0.html use.
> 
> As to the -dev etc. suffix(es) it's best generated in -nightly, -pr etc. automated "tagged" release-builds rather than off manifest / build.rs which should reflect the "base version" at master / main or the expected future release.

I agree with the base version here, but was under the impression that people were leaning towards this solution.

> 
> e.g. look nightly release builder:
> 
> https://github.com/pinkforest/test-auditable-releases
> 
> It makes nightly releases based on matrix using git tags pushing to timestmaped nightly ver.
> 
> https://github.com/pinkforest/test-auditable-releases/releases/tag/cargo-auditable%40nightly-20220830-0324
> 
> I have a separate repo for radicle-nightly but I'm tuning some of the target triplets that I need for testing:
> 
> https://github.com/pinkforest/rad-release/releases/tag/radicle-cli%40nightly-20220908-0511
> 
> When the CHANGELOG.md is using standardised form typical release tooling has pattern to extract it out automatically.

Cool, that's very valuable! I'm in favor of providing nightly builds. But as mentioned above, it should be possible to distinguish local builds as well. What would be your proposal here?

**Comment by pinkforest** [comment]
>  fear that starting the conversation around nightly builds will block this from being done

It's already built...

> we'll lose CI for now.

They are portable workflows, not really, you can run them off github.

**Comment by slack-coder** [comment]
@pinkforest check the [issue](https://github.com/radicle-dev/radicle-cli/issues/239) for the PR, it may be a good place to follow up so as not to fragment the discussion.

**Comment by slack-coder** [comment]
```
rad --version
# rad 0.7.0-dev+10dfad2
```
I like it :)

**Pull request merged**